src/resty/ljsonschema/init.lua

```diff
@@ -713,6 +713,54 @@ generate_validator = function(ctx, schema)
       ctx:stmt(sformat('    return false, %s([[failed to match pattern ]] .. %q .. [[ with %%q]], %s)', ctx:libfunc('string.format'), format_escaped_pattern, ctx:param(1)))
       ctx:stmt(        '  end')
     end
+    if schema.format then
+      local pattern
+
+      if schema.format == "date" then
+        --[[
+          Spec: https://tools.ietf.org/html/rfc3339#section-5.6
+
+          JSON schema validation does not require section 5.7 validation
+          allowing for invalid date (e.g. 2020-04-31, and 2020-02-29).
+        ]]--
+        pattern = [[^(?<fullyear>\d{4})-(?<month>0[1-9]|1[0-2])-(?<mday>0[1-9]|[12][0-9]|3[01])$]]
+      elseif schema.format == "date-time" then
+        --[[
+          Spec: https://tools.ietf.org/html/rfc3339#section-5.6
+
+          JSON schema validation does not require section 5.7 validation
+          allowing for invalid date (e.g. 2020-04-31T00:00:00.000Z, and 2020-02-29T00:00:00.000Z).
+        ]]--
+        pattern = [[^(?<fullyear>\d{4})-(?<month>0[1-9]|1[0-2])-(?<mday>0[1-9]|[12][0-9]|3[01])[Tt](?<hour>[01][0-9]|2[0-3]):(?<minute>[0-5][0-9]):(?<second>[0-5][0-9]|60)(?<secfrac>\.[0-9]+)?([Zz]|(\+|-)(?<offset_hour>[01][0-9]|2[0-3]):(?<offset_minute>[0-5][0-9]))$]]
+      elseif schema.format == "time" then
+        -- Spec: https://tools.ietf.org/html/rfc3339#section-5.6
+        pattern = [[^(?<hour>[01][0-9]|2[0-3]):(?<minute>[0-5][0-9]):(?<second>[0-5][0-9]|60)(?<secfrac>\.[0-9]+)?([Zz]|(\+|-)(?<offset_hour>[01][0-9]|2[0-3]):(?<offset_minute>[0-5][0-9]))$]]
+      end
+
+      if pattern then
+        ctx:stmt(sformat('  if not %s(%s, %q) then', ctx:libfunc('custom.match_pattern'), ctx:param(1), pattern))
+        ctx:stmt(sformat('    return false, %s([[expected valid ]] .. %q .. [[, got %%q]], %s)', ctx:libfunc('string.format'), schema.format, ctx:param(1)))
+        ctx:stmt(        '  end')
+
+        -- Handle restrictions for dates; see spec https://tools.ietf.org/html/rfc3339#section-5.7
+        if string.sub(schema.format, 1, 4) == "date" then
+          ctx:stmt(sformat('  local fullyear, month, day = %s:match(%q)', ctx:param(1), "(%d+)-(%d+)-(%d+).*"))
```

Comment:
above we use the regexes, here we do it again, using captures and `string.match`, instead we should use `string.match` above, and re-use the captures here for further validation. That skips 1 matching operation.